mlir/lib/Dialect/MIOpen/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -41,53 +48,66 @@ struct LowerMIOpenOpsStep2Pass
   void runOnOperation() override;
 };
 
+// TODO(kdrewnia): Could rank-0 vectors clear some of this up?
+// Utility function for crafting optional vector types
+Type vectorTypeOrSelf(Type elementType, int64_t len) {
+  if (len == 1)
+    return elementType;
+  return VectorType::get({len}, elementType);
+}
+
 //===----------------------------------------------------------------------===//
 // Utility function to determine the type to be loaded
 //===----------------------------------------------------------------------===//
-template <typename T>
-std::tuple<Type, int, int, int>
-computeLoadStoreTypeInfo(OpBuilder &b, T &gop, Type elementType,
-                         SmallVectorImpl<Type> &variadicTypes, ArrayAttr dims,
-                         bool isMatrixA, int64_t KPack = 1) {
-  uint64_t loadLength = 1;
-  uint64_t storeLength = 1;
-  int vectorDim = dims.size() - 1;
-  if (isMatrixA) {
-    loadLength = gop->getAttr("matrix_a_source_data_per_read")
-                     .template cast<IntegerAttr>()
-                     .getInt();
-    storeLength = gop->getAttr("matrix_a_dest_data_per_write_dim_m")
-                      .template cast<IntegerAttr>()
-                      .getInt();
-    vectorDim = gop->getAttr("matrix_a_source_vector_read_dim")
-                    .template cast<IntegerAttr>()
-                    .getInt();
-  } else {
-    loadLength = gop->getAttr("matrix_b_source_data_per_read")
-                     .template cast<IntegerAttr>()
-                     .getInt();
-    storeLength = gop->getAttr("matrix_b_dest_data_per_write_dim_n")
-                      .template cast<IntegerAttr>()
-                      .getInt();
-    vectorDim = gop->getAttr("matrix_b_source_vector_read_dim")
-                    .template cast<IntegerAttr>()
-                    .getInt();
-  }
+void computeLoadStoreTypeInfo(OpBuilder &b, ArrayRef<int64_t> sliceLengths,
```

Comment:
The inputs and outputs are mixed up in the parameter list. Could we rearrange them, or fallback to using tuples?

Reply:
The outputs all come after the inputs (with the minor exception of `vectorDim` which is read-write to allow for KPack stuff.